### PR TITLE
Carets - Guillermina

### PR DESCRIPTION
**Pull request by murog**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? |  I read through the documentation and found an example on how to conduct a query using the word "chicken."  I played with the parameters on the link to figure out how to get what I needed out of the API. |
| Describe your API Wrapper.  How did you decide on the methods you created? |My API Wrapper only has two methods.  Reviewing them, I could have created some helper methods to DRY up the code since they are kind of long.  The first one  "self.list_recipes" takes in a params hash and pulls information it needs out of it to make a request.  If a "from" and "to" range is not include, it makes the query starting a 0 and ending at 10.  It also updates the url if a health label is given in the params.  If the query is successful, it returns an array of recipe objects.  If the query is unsuccessful, it returns an empty array.  The second method, "self.find_recipe", takes in a recipe uri (as a string) and makes a request based on the uri. If it is successful, it will return a recipe object.  If it is not, it returns false. |
| Describe an edge case or failure case test you wrote for your API Wrapper. | An edge I tested for was that it would be able to successfully make a request on a multi word query that includes blank spaces.  It was successful, which surprised me!   |
| Explain how VCR aids in testing an API. |  VCR saves requests you make on an API.  When you make a duplicate requests, it will compare your test again the saved response from the past.    This is helpful because it minimizes the amount of calls you make on an API and reduces the chances of you getting timed out or banned from using their service. |
| What is the Heroku URL of your deployed application? |  http://gm-muncher.herokuapp.com/|

### CODE REVIEW

File: test/controllers/recipes_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  describe "show" do
+    it "can get a show page for a given recipe" do
+      VCR.use_cassette("show_action") do
+        vegatable_stew_uri = "e128b2dfab86d911618b669d499c4274"
+        get show_recipe_path(vegatable_stew_uri)
+        must_respond_with :success
+      end
+    end
+
+    it "will render 404 for a bogus recipe" do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:29]: I would also run a test with a search term that gets no responses.

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good number of commits and good commit messages |
| Comprehension questions	| Check |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check |
| Semantic HTML | Check  |
| Errors are reported to the user | No error reported |
| API Wrapper to handle the API requests | Check |
| Controller testing | Check |
| Lib testing | Check |
| Search Functionality | Check |
| List Functionality | Check  |
| Show individual item functionality (link to original recipe opens in new tab) |  |
| **Styling** |  |
| Foundation Styling for responsive layout | Responsive, but the tables aren't great for smaller screens. |
| List View shows 10 items at a time/pagination | Check, but the pagination is there even for 0 results. |
| The app is styled to create an attractive user interface | Nicely attractive. |
| **API Features** | |
| The App attributes Edaman | Check |
| The VCR casettes do not contain the API key | Check  |
| **External Resources** |  |
| Link to deployed app on Heroku | Check |
|  **Overall** | Great work on the Google OAuth!  Well done, you hit all the major requirements.  Awesomework!   |
